src/pages/resource/[id].tsx

```diff
@@ -121,13 +121,11 @@ export const SingleResourcePage: NextPage<{ id: string }> = ({ id }) => {
   const { data: session } = useSession();
   const { user } = session ?? {};
 
-  if (isLoading) {
-    return <LoadingPage />;
-  }
+  if (isLoading) return <LoadingPage />;
 
-  if (!resource) {
-    return <div>404</div>;
-  }
+  if (!resource) return <div>404</div>;
+
+  if (resource.draft && resource.createdById !== user?.id) return <div>403 - Forbidden</div>
```

Comment:
This kind of check should happen only in the API (because otherwise we're giving users the possibility to still download the resource but just not showing it to them in the UI)

Reply:
understood, i think the resource api already covers this scenario anyway